IG Timer Update

Hilimii opened the pull request:
I've added a setting, for the campaign mode, which pauses the timer unless the player is in mission. It functions quite closely to the way AC7 record their IGT. This is, of course, not legal for runs, but I'm adding it so people can play around with it.

I plan to test and then push this Wednesday next week. I know you're taking a hiatus, so only respond if you've got the time! It's a pretty small and watertight update (hopefully).

Conditions which cause the timer to pause when using IGT setting:

- Player pawn is not defined
- Player is not in level (IntoLevel = 0)
- Player has completed the mission (missionComplete =3), but hasn't skipped ahead yet.
- Player has paused the mission

How this works in practice:

- Timer is paused completely outside of missions in all other sequences. This includes briefings, debriefings, landings, takeoffs, menus, and hangars
- Timer is paused until 1.0s after the player presses 'start' to begin a mission
- Timer is immediately paused when the player resets a mission, and only resumed when the above bullet point becomes true
- Timer is paused when the player pauses in-game

Hilimii committed:
Adds IG Timer Setting and Conditions

Added a new setting for an IGT mode when running campaign.
Added intoLevel into pointer list.

Hilimii committed:
Updated version info at top

jwfraustro commented:
I have been taking a bit of a hiatus, but seeing as you've broken a few of my records I'm tempted to get back in the game. :)

This looks good to me and I trust your ability to test it. 

I somewhat wish we could separate these different timing rules into separate files, just for the purposes of keeping the logic contained, but I don't think that's possible.

Thanks for doing this! I wish this was the way the record timing actually worked!

Hilimii committed:
Fixed Setting Bug

Caught a bug which caused the IGT setting to work when not toggled on. Added some brackets to ensure it does not trigger when unintended.

Hilimii commented:
Yeeee I'm glad you're thinking of getting back into it! :) I've been having a blast using the autosplitter to claim all those records 😅

I caught a logic bug and fixed it. It works perfectly now!

And yeahhhh the logic is nearing the point of becoming a headache, despite my best efforts to keep it organised, but c'est la vie.
